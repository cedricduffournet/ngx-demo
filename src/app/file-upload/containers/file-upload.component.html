<div class="message">
  <input #file type="file" (change)="uploadFile($event)" accept="image/png, image/jpeg">
</div>