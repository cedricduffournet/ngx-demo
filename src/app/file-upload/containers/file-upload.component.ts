import { Component, OnInit } from '@angular/core';
import { select, Store } from '@ngrx/store';
import { Observable } from 'rxjs';

import * as fromFileUpload from '@app/file-upload/state/reducers';
import { FileUploadActions } from '@app/file-upload/state/actions';
// import * as fromFileUploadSelectors from 'src/app/upload-file-store/selectors';
// import * as fromFileUploadState from 'src/app/upload-file-store/state';

@Component({
  selector: 'app-file-upload',
  templateUrl: './file-upload.component.html'/* ,
  styleUrls: ['./file-upload.component.css'] */
})
export class FileUploadComponent implements OnInit {
  completed$: Observable<boolean>;
  progress$: Observable<number | null>;
  error$: Observable<string | null>;
  isInProgress$: Observable<boolean>;
  isReady$: Observable<boolean>;
  hasFailed$: Observable<boolean>;

  constructor(private store: Store<fromFileUpload.State>) {}

  ngOnInit() {
    /*     this.completed$ = this.store.pipe(
      select(fromFileUploadSelectors.selectUploadFileCompleted)
    ); */

    this.progress$ = this.store.pipe(
      select(fromFileUpload.getFileUploadProgress)
    );

    this.error$ = this.store.pipe(select(fromFileUpload.getFileUploadError));

    /*     this.isInProgress$ = this.store$.pipe(
      select(fromFileUploadSelectors.selectUploadFileInProgress)
    );

    this.isReady$ = this.store$.pipe(
      select(fromFileUploadSelectors.selectUploadFileReady)
    );

    this.hasFailed$ = this.store$.pipe(
      select(fromFileUploadSelectors.selectUploadFileFailed)
    ); */
  }

  uploadFile(event: any) {
    const files: FileList = event.target.files;
    const file = files.item(0) as File;
    console.log(file);
    const sendFile = {
      ...file,
      name: file.name
    };

    console.log(sendFile);
    /*     const files: FileList = event.target.files;
    
 */


    this.store.dispatch(FileUploadActions.uploadRequest({file}));
/*       new fromFileUploadActions.UploadRequestAction({
        file
      })
    ); */

    // clear the input form
    event.srcElement.value = null;
  }

  resetUpload() {
    this.store.dispatch(FileUploadActions.uploadReset() );
    // this.store$.dispatch(new fromFileUploadActions.UploadResetAction());
  }

  cancelUpload() {
    this.store.dispatch(FileUploadActions.uploadCancel() );
    // this.store$.dispatch(new fromFileUploadActions.UploadCancelAction());
  }
}
