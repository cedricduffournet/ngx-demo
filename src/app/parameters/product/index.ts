export * from './parameter-product.component';
export * from './parameter-product-add.component';
export * from './parameter-product-details.component';
