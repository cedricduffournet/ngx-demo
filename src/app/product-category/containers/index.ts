export * from './product-category-add-modal/product-category-add-modal.component';
export * from './product-category-update-modal/product-category-update-modal.component';
export * from './product-category-delete-modal/product-category-delete-modal.component';
export * from './product-category-list-view/product-category-list-view.component';
