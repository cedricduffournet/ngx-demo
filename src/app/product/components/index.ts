export * from './product-add/product-add.component';
export * from './product-update/product-update.component';
export * from './product-delete/product-delete.component';
export * from './product-form/product-form.component';
export * from './product-item/product-item.component';
export * from './product-items/product-items.component';
export * from './product-details/product-details.component';
