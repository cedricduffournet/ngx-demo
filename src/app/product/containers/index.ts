export * from './product-add-view/product-add-view.component';
export * from './product-update-modal/product-update-modal.component';
export * from './product-delete-modal/product-delete-modal.component';
export * from './product-list-view/product-list-view.component';
export * from './product-selected/product-selected.component';
export * from './product-details-view/product-details-view.component';
