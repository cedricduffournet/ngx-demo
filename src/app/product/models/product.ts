import { schema } from 'normalizr';
import {
  ProductCategory,
  productCategorySchema
} from '@app/product-category/models/product-category';

export interface ProductBase {
  id: number;
  name: string;
  description: string;
  priceAmount: number;
}

export interface Product extends ProductBase {
  categories: number[];
}

export interface ProductFull extends ProductBase {
  categories: ProductCategory[];
}

export const productSchema = new schema.Entity('products', {
  categories: [productCategorySchema]
});

export const getPrice = (price: number) => {
  if (price > 0) {
    return price / 100;
  }
  return 0;
};
