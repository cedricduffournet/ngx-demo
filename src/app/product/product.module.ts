import { NgModule, ComponentFactoryResolver } from '@angular/core';

import { RouterModule } from '@angular/router';
import { EffectsModule } from '@ngrx/effects';
import { StoreModule } from '@ngrx/store';

import { SharedModule } from '@app/shared/shared.module';
import { ProductEffects } from '@app/product/state/effects';
import { ProductFacade } from '@app/product/state/product.facade';
import {
  ProductAddComponent,
  ProductUpdateComponent,
  ProductDeleteComponent,
  ProductFormComponent,
  ProductItemComponent,
  ProductItemsComponent
} from '@app/product/components';
import {
  ProductAddModalComponent,
  ProductUpdateModalComponent,
  ProductDeleteModalComponent,
  ProductListViewComponent
} from '@app/product/containers';
import { reducers } from '@app/product/state/reducers';
import { CoalescingComponentFactoryResolver } from '@app/coalescing-component-factory-resolver.service';

@NgModule({
  imports: [
    RouterModule,
    SharedModule,
    EffectsModule.forFeature([ProductEffects]),
    StoreModule.forFeature('products', reducers)
  ],
  declarations: [
    ProductListViewComponent,
    ProductAddComponent,
    ProductUpdateComponent,
    ProductDeleteComponent,
    ProductFormComponent,
    ProductItemComponent,
    ProductItemsComponent,
    ProductAddModalComponent,
    ProductUpdateModalComponent,
    ProductDeleteModalComponent
  ],
  entryComponents: [
    ProductUpdateModalComponent,
    ProductAddModalComponent,
    ProductDeleteModalComponent
  ],
  exports: [ProductListViewComponent],
  providers: [ProductFacade]
})
export class ProductModule {
  // see https://github.com/angular/angular/issues/14324
  constructor(
    coalescingResolver: CoalescingComponentFactoryResolver,
    localResolver: ComponentFactoryResolver
  ) {
    coalescingResolver.registerResolver(localResolver);
  }
}
