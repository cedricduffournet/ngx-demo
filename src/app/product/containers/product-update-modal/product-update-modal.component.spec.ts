import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';

import { of } from 'rxjs';
import { BsModalRef } from 'ngx-bootstrap/modal';
import { provideMockStore } from '@ngrx/store/testing';
import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { ValidationActionModule } from '@app/shared/validation-action';
import { ModalWrapperModule } from '@app/shared/modal';
import { ProductFacade } from '@app/product/state/product.facade';

import {
  ProductUpdateComponent,
  ProductFormComponent
} from '@app/product/components';
import { ProductUpdateModalComponent } from '@app/product/containers';
import { Product } from '@app/product/models/product';

describe('UpdateProductModalComponent', () => {
  let fixture: ComponentFixture<ProductUpdateModalComponent>;
  let component: ProductUpdateModalComponent;
  let facade: ProductFacade;
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [
        ProductUpdateComponent,
        ProductUpdateModalComponent,
        ProductFormComponent
      ],
      imports: [
        TranslateModule.forRoot(),
        FormModule,
        ValidationActionModule,
        ModalWrapperModule,
        ReactiveFormsModule
      ],
      providers: [
        provideMockStore(),
        BsModalRef,
        {
          provide: ProductFacade,
          useValue: {
            selected$: of({
              name: 'TestName'
            }),
            updated$: of(false),
            updateProduct: jest.fn()
          }
        }
      ]
    });

    fixture = TestBed.createComponent(ProductUpdateModalComponent);
    component = fixture.componentInstance;
    facade = TestBed.get(ProductFacade);
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should call updateProduct event on submit', () => {
    spyOn(facade, 'updateProduct');
    const data = {
      id: 1,
      product: {
        name: 'name'
      } as Product
    };
    fixture.detectChanges();
    component.onUpdate(data);
    expect(facade.updateProduct).toHaveBeenCalledWith(data);
  });

  it('should close if product updated', () => {
    spyOn(component.bsModalRef, 'hide');
    facade.updated$ = of(true);
    fixture.detectChanges();
    expect(component.bsModalRef.hide).toHaveBeenCalled();
  });

  it('should close modal on cancel', () => {
    spyOn(component.bsModalRef, 'hide');
    fixture.detectChanges();
    component.onCancel();
    expect(component.bsModalRef.hide).toHaveBeenCalled();
  });

  it('should unsubscribe subscription when destroyed', () => {
    fixture.detectChanges();
    spyOn(component.subscription, 'unsubscribe');
    component.ngOnDestroy();
    expect(component.subscription.unsubscribe).toHaveBeenCalled();
  });
});
