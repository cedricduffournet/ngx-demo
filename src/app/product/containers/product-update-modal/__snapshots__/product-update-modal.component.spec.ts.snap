// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateProductModalComponent should be created 1`] = `
<app-product-update-modal
  bsModalRef={[Function BsModalRef]}
  categories$={[Function Observable]}
  facade={[Function Object]}
  loadingCategories$="undefined"
  productCategoryFacade={[Function Object]}
  selectedProduct$={[Function Observable]}
  subscription={[Function Subscriber]}
  updated$={[Function Observable]}
  updating$="undefined"
>
  <app-modal-wrapper
    ng-reflect-title="EDIT_PRODUCT_TITLE"
  >
    <div
      class="modal-content bg-light"
    >
      
      <div
        class="modal-header"
      >
        <h4
          class="modal-title"
        >
          EDIT_PRODUCT_TITLE
        </h4>
      </div>
      <div
        class="modal-body p-0"
      >
        
      </div>
    </div>
  </app-modal-wrapper>
</app-product-update-modal>
`;
